[Merged by Bors] - fix go vet

noamnelke opened the pull request:
## Motivation
`go vet ./...` reports some errors.

## Changes
Fix `go vet` errors.
One issue was an actual bug where we'd call `continue` in a loop instead of performing the rest of the loop.
Deleted some zombie code as well.

noamnelke committed:
fix go vet

antonlerner commented on activation/activation_test.go:
```diff
@@ -51,7 +51,7 @@ var (
 	npst             = NewNIPSTWithChallenge(&chlng, poetRef)
 	commitment       = &types.PostProof{
-		//Identity:     []byte(nil),
+		// Identity:     []byte(nil),
```
remove commment?

noamnelke replied:
Sure, I just used the IDE's auto-format, didn't actually look at every line, will remove all of these now.

antonlerner commented on activation/activationdb_test.go:
```diff
@@ -201,7 +172,7 @@ func TestATX_ActiveSetForLayerView(t *testing.T) {
 	atxdb, layers, _ := getAtxDb(t.Name())
 	blocksMap := make(map[types.BlockID]struct{})
-	//layers.AtxDB = &AtxDbMock{make(map[types.ATXID]*types.ActivationTx), make(map[types.ATXID]*types.NIPST)}
+	// layers.AtxDB = &AtxDbMock{make(map[types.ATXID]*types.ActivationTx), make(map[types.ATXID]*types.NIPST)}
```
remove comment?

antonlerner commented on activation/activationdb_test.go:
```diff
@@ -224,7 +195,7 @@ func TestATX_ActiveSetForLayerView(t *testing.T) {
 		assert.NoError(t, err)
 		atx.Nipst = NewNIPSTWithChallenge(hash, poetRef)
-		//layers.AtxDB.(*AtxDbMock).AddAtx(atx.ID(), atx)
+		// layers.AtxDB.(*AtxDbMock).AddAtx(atx.ID(), atx)
```
same?

antonlerner commented on miner/builder_test.go:
```diff
@@ -556,7 +535,7 @@ func TestBlockBuilder_getVotes(t *testing.T) {
 	n1 := service.NewSimulator().NewNode()
 	allblocks := []*types.Block{b1, b2, b3, b4, b5, b6, b7}
-	//st := []types.BlockID{b1.ID(), b2.ID(), b3.ID()}
+	// st := []types.BlockID{b1.ID(), b2.ID(), b3.ID()}
```
remove?

antonlerner commented on miner/builder_test.go:
```diff
@@ -129,7 +108,7 @@ func TestBlockBuilder_StartStop(t *testing.T) {
 	beginRound := make(chan types.LayerID)
 	n := net.NewNode()
-	//receiver := net.NewNode()
+	// receiver := net.NewNode()
```
remove?

y0sher reviewed (approved):
you changed some value receivers to be pointer receivers was that a `go vet` thing ?

noamnelke commented:
> you changed some value receivers to be pointer receivers was that a `go vet` thing ?

Yes, it warns that we pass a mutex by value.

e.g.:
```
activation/atxdb.go:570: getTopAtx passes lock by value: activation.ActivationDb
```

noamnelke committed:
PR fixes

noamnelke commented:
bors merge

y0sher commented:
ok. though we could also change the mutex to be of type pointer..

noamnelke commented:
Yes, but it's a bit more involved. A pointer to a mutex needs to be initialized (it starts out being `nil`), while making the method receiver a pointer is more straightforward.

I wonder why you resist this... Is it because using a pointer receiver implies that the method mutates the struct? In all of the places that I changed all other methods, except the one I changed, had pointer receivers, so it's not like the struct is supposed to be immutable.

noamnelke commented:
bors merge
